wiki/common-platforms/ros/ros-unsupported-os.md

```diff
@@ -0,0 +1,184 @@
+---
+# Jekyll 'Front Matter' goes here. Most are set by default, and should NOT be
+# overwritten except in special circumstances. 
+# You should set the date the article was last updated like this:
+date: 2023-05-03 # YYYY-MM-DD
+# This will be displayed at the bottom of the article
+# You should set the article's title:
+title: How to run a ROS distribution on an operating system that doesn't natively support it?
+# The 'title' is automatically displayed at the top of the page
+# and used in other parts of the site.
+---
+Each ROS distribution supports only a select number of platforms, and for all supported platforms, readily available Debian packages are available which can be easily installed using package managers such as apt. However, there often arise situations where you need to install and use a ROS distribution on a platform that is not supported for the distribution. This issue is a common occurrence when attempting to run specific ROS distributions on Single Board Computers (SBCs). The problem often arises due to the SBC's published images being either too outdated or too recent to support the required ROS distribution. This article will describe how to install and run ROS distributions in such cases. 
+
+## Approaches
+### 1. Running ROS2 Humble in a docker container on Ubuntu 20.04
+- Pros:
+  - Easy Setup
+  - Isolated Environment
+- Cons:
+  - Limited control
+  - Performance Overheads
+### 2. Installing ROS2 Humble from source on Ubuntu 20.04
+- Pros:
+  - More Control
+  - Better Performance
+- Cons:
+  - Time-consuming
+  - Difficult to maintain
+
+
+## Setups
+### 1. Running ROS2 Humble via a docker container on Ubuntu 20.04
+
+#### Installing Docker
+Follow instructions given here to install docker - https://roboticsknowledgebase.com/wiki/tools/docker/
+#### Running ROS2 Humble docker container
+Running the command below will pull the latest ROS2 Humble Docker Image, start a container and attach a shell to it
+```
```

Comment:
Specify language for nicer, syntax-highlighted snippets.
```suggestion
```sh
```

Reply:
Added this change